### PR TITLE
LOG-3499: add /var/log/oauth-server/audit.log to the log collection path

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### Description
This PR:
- updates `fluentd` config to collect logs from `/var/log/oauth-server/audit.log`
- updates `vector` config to collect logs from `/var/log/oauth-server/audit.log`
- updates corresponding unit tests 

/cc 
/assign 

/cherry-pick 

### Links

- Depending on PR(s):
- Bugzilla:
- Github issue:
- JIRA: https://issues.redhat.com/browse/LOG-3499
- Enhancement proposal:

### CODE REVIEW

File: internal/generator/fluentd/conf_test.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ var _ = Describe("Testing Complete Config Generation", func() {
   @type tail
   @id openshift-audit-input
   @label @INGRESS
-  path /var/log/oauth-apiserver/audit.log,/var/log/openshift-apiserver/audit.log
+  path /var/log/oauth-apiserver/audit.log,/var/log/openshift-apiserver/audit.log,/var/log/auth-server/audit.log
```
Comment (vimalk78) [inline internal/generator/fluentd/conf_test.go:205]: it should be `/var/log/oauth-server/audit.log` as per Jira comment

**Review by vimalk78 — CHANGES_REQUESTED** [review]
Check last comment in Jira task

**Review by jcantrill — COMMENTED** [review]
/hold
/approve

**Comment by vimalk78** [comment]
if possible in this PR (or in separate PR), 
- update `test/helpers/types/types.go` to add a struct for the structure of this new audit log
- update data model for vector audit logs in `internal/generator/vector/normalize.go`

File: internal/generator/fluentd/conf_test.go (hunk 1 above)
Comment (vparfonov) [reply internal/generator/fluentd/conf_test.go:205]: eagle eye

**Comment by vparfonov** [comment]
> if possible in this PR (or in separate PR),
> 
> * update `test/helpers/types/types.go` to add a struct for the structure of this new audit log
> * update data model for vector audit logs in `internal/generator/vector/normalize.go`

@vimalk78 What's so special in log from `oauth-server` that we might need a separate type for it? It still can be [OpenshiftAuditLog](https://github.com/vparfonov/cluster-logging-operator/blob/549d150a181e3e2ebd820bc78553d87209530bbb/test/helpers/types/types.go#L419), no?

**Comment by vimalk78** [comment]
> @vimalk78 What's so special in log from `oauth-server` that we might need a separate type for it? It still can be [OpenshiftAuditLog](https://github.com/vparfonov/cluster-logging-operator/blob/549d150a181e3e2ebd820bc78553d87209530bbb/test/helpers/types/types.go#L419), no?

I am not sure about the format of these logs. if it is same as OpenshiftAuditLogs, yes we can create a type alias and use same.

**Comment by jcantrill** [comment]
> > @vimalk78 What's so special in log from `oauth-server` that we might need a separate type for it? It still can be [OpenshiftAuditLog](https://github.com/vparfonov/cluster-logging-operator/blob/549d150a181e3e2ebd820bc78553d87209530bbb/test/helpers/types/types.go#L419), no?
> 
> I am not sure about the format of these logs. if it is same as OpenshiftAuditLogs, yes we can create a type alias and use same.

Per issue comment from @vparfonov its a  kube event

**Review by jcantrill — COMMENTED** [review]
/lgtm
/hold cancel

**Pull request merged**